### DISCUSSION TITLE
What Project Goals are important for you

### DISCUSSION THREAD

**Discussion by Flyingmana** [OpenMage/magento-lts#1354] [General]
While we as Maintainers have certain Notions what the Goals of the project are, they may differ or not cover all of the Goals the Community is seeing in OpenMage. And of Course its not possible to make everyone happy.
But for a healthy relationship with the Community it is a valuable Tool to regularly check if the Goals still align or if they maybe need a bit of modification.

So, what things would you put on to the List of Goals. Not only what you think the project should do, but also what you would like to see the project do.

**Comment by Bashev** [comment]
Keep magento 1 secure (as is possible);

**Comment by joshua-bn** [comment]
For me, this project helps me track issues with what we have running in production. I don't use OM directly, but since it's based on M1 and so is my company's codebase, I use a lot of this.

That being said, security first, then stability, then performance. Don't care about new features. In fact, that just complicates things for me.

**Reply by joshuaadickerson** [reply]
One big thing I'd love to see - and I think everyone would agree - tests.

**Comment by addison74** [comment]
1. Solve security breaches.

2. Compatibility with the newest version of PHP, MySQL.

3. Better performance. Achieve the best performance by optimizing the code, SQL queries, server configuration. If we could use Varnish like in M2 it will be a huge step forward.

4. Functionality. Fix most of the bugs found in the code.

5. Improvement for the current features Magento 1 offers. For example, we cannot set positions for Terms & Conditions, in product grid Attribute Set column is not ordered ascending, when creating an attribute set the attributes on the right column are not ordered ascending, wrong attributes order in comparison window, cannot add to wishlist from tag page in customer account, deleting reviews leaves orphan data in database table, filter bill to name and ship to name in order grid allows only one word, and many more I reported over the years to M1 team.

**Comment by kiatng** [comment]
My client has relied on M1 for more than a decade. Over the years, we added 20 custom modules, which are still being actively maintained and modified. Each version bump in M1 required many months of implementation. As for migrating to M2, my estimation is that it'll take a couple of years. So OM is the natural choice. What I look for in OM are the same as what this project has been doing:

1. Security.
2. Bug fixes.
3. Improvement on current features.
4. Compatibility with future PHP8, 9, 10...

We do not have any issue with performance, especially after updating to PHP7.x, but even when it was PHP5.x, it was deemed to be OK. 

As for new features, it depends but not against. The question I have is that do we want to start new projects based on OM? I am implementing new OM projects, by reusing our custom modules that we developed over the years to quickly spin out to test and meet business requirements.  For these I implemented Vue.js, which is not compatible with the current prototype.js ver 1.7 in OM. But it plays nice with the latest 1.7.3. And we can still manage QR code with a legacy PHP module. But how far can OM go? I do not know if it's possible, but there may come a time when we need to interact with API offerings such as Google Machine Learning that requires installing new packages from Google. As I understand, javascript and new PHP symfony-like packages are not part of the OM project goals. So, in my opinion, why new features when there is a bunch of legacy stuff that can't be touched.

**Comment by tmotyl** [comment]
for me it's:
- "scratch my own itch"
- performance
- compat with newest php versions
- bugfixes and ease of use for developers (e.g. the fixes to doc comments improved the everyday life of the developers so much!)

I think in order to maintain openmage in the longer run, its very important to resurect the test suite.

**Comment by sreichel** [comment]
Keep M1 alive bit longer. For small/med business it (OM) is a good solution. :)

- fix year old bugs
- improve performance
- make it developer friendly (DOCs)
- clean code that follows best practice (follow code analysize tools)

**Comment by woutersamaey** [comment]
For me it’s:
- Making a great product even better by fixing things that Magento never cared about.
- Modernizing the stack for PHP7&8, so it can keep going indefinitely 
- Making my job as a developer more fun
- And maybe we can end up with a spin-off that has it’s own right to exist next to M2, which is honestly too complex and bulky for most SME and scale-ups.
